docs/plugins/npe2_migration_guide.md

```diff
@@ -1,39 +1,35 @@
 (npe2-migration-guide)=
 
-# npe2 migration guide
+# `npe2` migration guide
 
-```{warning}
-This guide is still a work in progress.
-
-There are inaccuracies and mistakes. If you notice any, feel free to submit
-issues to the [napari github repository](https://github.com/napari/napari).
-```
-
-We've introduced a new plugin engine. The new library [npe2] is a
-re-imagining of how napari interacts with plugins. Rather than importing a
-package to discover plugin functionality, a static manifest file is used to
-declaratively describe a plugin's capabilities. Details can be found in the
-[](npe2-manifest-spec).
+This document details how to convert a plugin using the first generation
+`napari-plugin-engine`, to the new `npe2` format.  
 
-Plugins targeting `napari-plugin-engine` will continue to work, but we
-recommend migrating to `npe2`. This guide will help you learn how!
+The primary difference between the first generation and second generation plugin
+system relates to how napari *discovers* plugin functionality. In the first
+generation plugin engine, napari had to *import* plugin modules to search for
+hook implementations decorated with `@napari_hook_implementation`. In `npe2`,
+plugins declare their contributions *statically* with a [manifest
+file](./manifest).
```

Comment:
Maybe put this in an admonition or a "why upgrade" section. 

I feel like this is a nice description of an important technical benefit npe2 - a reason someone should upgrade their plugin. We need something like this, and I think the migration guide is the right spot, but to me it reads more like a side-note than part of the main flow.

Reply:
+1 for a "Why upgrade" section

Reply:
Could also add something about how start up will be faster. I think ok to merge this PR without and then add a longer "Why upgrade" section in a followup PR